save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/component/CvssBaseScoreCalculator.kt

```diff
@@ -0,0 +1,335 @@
+                        setBaseMetrics { it.copy(availability = CiaType.NONE) }
+                    }
+                }
+            }
+        }
+
+        div {
+            className = ClassName("row mt-4")
+            div {
+                className = ClassName("col-12")
+
+                header("Severity score vector")
+
+                li {
+                    className = ClassName("list-group-item d-flex justify-content-center")
+                    +"${if (baseMetrics.isValid()) "${getCriticality(calculateScore(baseMetrics))} ${ calculateScore(baseMetrics) } " else ""}${baseMetrics.scoreVectorString()}"
+                }
+            }
+        }
+
+        div {
+            className = ClassName("mt-4 modal-footer")
+            buttonBuilder("Ok", "success", classes = "mr-2", isDisabled = !baseMetrics.isValid()) {
+                props.onCloseButtonPassed(baseMetrics)
+                props.onCloseButton()
+            }
+            buttonBuilder("Close", "secondary") {
+                props.onCloseButton()
+            }
+        }
+    }
+}
+
+/**
+ * [Props] for [cvssBaseScoreCalculator]
+ */
+external interface CvssBaseScoreCalculatorProps : Props {
+    /**
+     * Callback to close window
+     */
+    var onCloseButton: () -> Unit
+
+    /**
+     * Callback to update base metrics
+     */
+    var onCloseButtonPassed: (BaseMetrics) -> Unit
+}
+
+private fun ChildrenBuilder.radioButton(
+    isActive: Boolean,
+    buttonName: String = "",
+    activeStyle: String = "outline-dark",
+    isCustomColor: Boolean = false,
+    onClickFun: FormEventHandler<HTMLLabelElement>,
+) {
+    label {
+        className = ClassName("btn ${if (isActive && isCustomColor) {
+            ""
+        } else {
+            "btn-${if (isActive) "$activeStyle active" else "outline-dark"}"
+        }}")
+
+        style = jso {
+            if (isActive && isCustomColor) {
+                color = "#212529".unsafeCast<Color>()
+                backgroundColor = "#FFFF00".unsafeCast<BackgroundColor>()
+            }
+            borderColor = "#212529".unsafeCast<BorderColor>()
+        }
+
+        input {
+            name = "options"
+            type = InputType.radio
+            autoComplete = AutoComplete.off
+        }
+        onChange = onClickFun
+        +buttonName
+    }
+}
+
+private fun ChildrenBuilder.header(
+    name: String
+) {
+    dt {
+        className = ClassName("text-xs")
+        style = jso {
+            width = "100%".unsafeCast<Width>()
+            color = "#ffffff".unsafeCast<BorderColor>()
+            backgroundColor = "#404040".unsafeCast<BorderColor>()
+            borderColor = "#000000".unsafeCast<BorderColor>()
+        }
+        +name
+    }
+}
+
+private fun getCriticality(value: Float): String = when (value) {
+    in 0f..3.9f -> "Low"
+    in 3.9f..6.9f -> "Medium"
+    in 6.9f..8.9f -> "High"
+    in 8.9f..10f -> "Critical"
```

Comment:
https://ru.wikipedia.org/wiki/Common_Vulnerability_Scoring_System
I guess should be 
```suggestion
private fun getCriticality(value: Float): String = when (value) {
    0f -> "N/A"
    in 0.1f..3.9f -> "Low"
    in 4.0f..6.9f -> "Medium"
    in 7.0f..8.9f -> "High"
    in 9.0f..10f -> "Critical"
    else -> throw IllegalStateException("Progress should be in [0; 10.0], got $value")
}
```

Reply:
Changes